src/main/kotlin/no/nav/rekrutteringsbistand/stillingssokproxy/Security.kt

```diff
@@ -0,0 +1,95 @@
+import no.nav.security.token.support.core.http.HttpRequest
+import no.nav.security.token.support.core.validation.JwtTokenValidationHandler
+import java.net.URL
+import javax.servlet.http.Cookie
+
+class Security {
+
+    private val issuer_isso = "isso-idtoken"
+
+    fun lagSikkerhetsfilter(javalin: Javalin, tillateUrl: List<String>) {
+        javalin.before { context ->
+            val url: String = context.req.requestURL.toString()
+            val erÅpenUrl = tillateUrl.any { tillattUrl ->
+                url == tillattUrl
+            }
+
+            if (!erÅpenUrl) {
+                try {
+                    val veileder = innloggetVeileder(context)
+                    context.cookieStore("innloggetVeileder", veileder)
+                } catch (e: Exception) {
+                    context.status(403)
+                }
+            }
+        }
+    }
+
+    private fun innloggetVeileder(context: Context): InnloggetVeileder {
+        val tokenValidationHandler = JwtTokenValidationHandler(getMultiIssuerConfiguration())
+        val tokenValidationContext = tokenValidationHandler.getValidatedTokens(getHttpRequest(context))
+
+        val claims = tokenValidationContext.getClaims(issuer_isso)
+
+        val innloggetVeileder = InnloggetVeileder(
+            claims["unique_name"].toString(),
+            claims["name"].toString(),
+            claims["NAVident"].toString()
+        )
+        innloggetVeileder.validate()
+
+        log("Sikkerhetsfilter").info("InnloggetVeileder: $innloggetVeileder")
+        return innloggetVeileder
+    }
+
+    private fun getMultiIssuerConfiguration(): MultiIssuerConfiguration {
+        val properties = when (System.getenv("NAIS_CLUSTER_NAME")) {
+            "dev-gcp" -> IssuerProperties(
+                URL("https://login.microsoftonline.com/NAVQ.onmicrosoft.com/.well-known/openid-configuration"),
+                listOf("38e07d31-659d-4595-939a-f18dce3446c5"),
+                issuer_isso
+            )
+            "prod-gcp" -> IssuerProperties(
+                URL("https://login.microsoftonline.com/navno.onmicrosoft.com/.well-known/openid-configuration"),
+                listOf("9b4e07a3-4f4c-4bab-b866-87f62dff480d"),
+                issuer_isso
+            )
+            else -> throw RuntimeException("Ukjent cluster")
+        }
+
+        return MultiIssuerConfiguration(mapOf(Pair(issuer_isso, properties)))
+    }
+
+    private fun getHttpRequest(context: Context): HttpRequest = object : HttpRequest {
+        override fun getHeader(headerName: String?) = context.headerMap()[headerName]
+        override fun getCookies() =
+            context.cookieMap().map { (name, value) -> NameValueImpl(Cookie(name, value)) }.toTypedArray()
+    }
+
+    private class NameValueImpl(val cookie: Cookie) : HttpRequest.NameValue {
+        override fun getName() = cookie.name
+        override fun getValue() = cookie.value
+    }
```

Comment:
Har vi behov for denne klassen?
Skjønner ikke helt hva den gjør... 😄

Reply:
Den brukes for å lage et HttpRequest-objekt (av en klasse fra token-support biblioteket) som trenger en liste av slike objekter. Klassen blir brukt i metoden over. I bibliotekets core-del finnes NameValue kun som interface.

Reply:
I biblioteket er den implementert anonymt, som en privat klasse vi ikke har tilgang til og som en offentlig klasse i ktor-modulen som vi ikke bruker.

Reply:
Vi kan implementere den anonymt, men må da gjøre det inne i en stream, og syns det ser litt rotete ut da..

Reply:
Vi har gjort det anonymt nå!